tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal

```diff
@@ -519,3 +519,35 @@ table<BarRec> key(x, y, z) tb3 = table [
     {x: i, y: i, z: "a"},
     {...spreadField3}
 ];
+
+type Employee2 record {
+    readonly int id;
+    string name;
+};
+
+function testKeyConstraint() {
+    table<Employee2> superTable1 = table key(id) [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+
+    table<Employee2> superTable2 = table [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+
+    table<Employee2> key<int> keyTable = table key(id) [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+
+    table<Employee2> key<never> keylessTable = table [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+
```

Comment:
Test where keyless table table<Type> key<never> assigned to table<Type>

Reply:
type is `table<Type> key<never>`
```
table<Employee2> key<never> keylessTable = table [
        {id: 0, name: "a"},
        {id: 1, name: "b"}
    ];

    _ = keylessTable.remove(0);
```
But for this test case error msg is `incompatible types: expected 'never', found 'int'`. Is that error msg is correct or it should like `'remove' method is not allowed for keyless table`?

Reply:
That's wrong right. Since the table constructor does not have a key specifier/constraint; it should not get an incompatible error. Where does this `int` comes from?

Reply:
Actually error occurs at `_ = keylessTable.remove(0);` the `int` is `0`.

Reply:
That's also wrong. We should not allow calling "remove" for keyless table